src/libponyrt/mem/heap.c

```diff
@@ -524,7 +525,12 @@ void ponyint_heap_used(heap_t* heap, size_t size)
 
 bool ponyint_heap_startgc(heap_t* heap)
 {
-  if(heap->used <= heap->next_gc)
+  // don't run GC if there's no heap used
+  if(heap->used == 0)
```

Comment:
is this really needed?

Reply:
yes and no. the check for `heap->used == 0` needs to be done for when `ponyint_pool_mem_pressure() == true`. I put it as a separate thing but I can instead add it later on as part of the `ponyint_pool_mem_pressure()` condition.

Reply:
Ah, right that makes sense.

Reply:
If this were to remain as is, I think an explanation of why it's needed would be important. This also makes me favor the "do this as capping next_gc" approach I put forth.